python/ray/tests/test_memory_deadlock.py

```diff
@@ -246,18 +246,66 @@ def test_deadlock_two_sets_of_task_with_nested_task(
         - 2 * parent_bytes
     )
 
-    barrier = BarrierActor.options(
-        max_restarts=0, max_task_retries=0, max_concurrency=2
-    ).remote(2)
+    ref1 = task_with_nested_task.options(max_retries=-1).remote(
+        parent_bytes, nested_bytes
+    )
+    ref2 = task_with_nested_task.options(max_retries=-1).remote(
+        parent_bytes, nested_bytes
+    )
 
-    ref1 = task_with_nested_task.options(max_retries=0).remote(
-        parent_bytes, nested_bytes, barrier
+    with pytest.raises(ray.exceptions.GetTimeoutError) as _:
+        ray.get(ref1, timeout=120)
+    with pytest.raises(ray.exceptions.GetTimeoutError) as _:
+        ray.get(ref2, timeout=120)
+
+
+@pytest.mark.skipif(
+    sys.platform != "linux" and sys.platform != "linux2",
+    reason="memory monitor only on linux currently",
```

Comment:
parametrize the fixture and expectation

https://docs.pytest.org/en/latest/example/parametrize.html#apply-indirect-on-particular-arguments